x-pack/plugins/security_solution/public/detection_engine/rule_details_ui/pages/rule_details/index.tsx

```diff
@@ -610,9 +612,14 @@ const RuleDetailsPageComponent: React.FC<DetectionEngineComponentProps> = ({
                             !hasUserCRUDPermission(canUserCRUD) ||
                             (isMlRule(rule?.type) && !hasMlPermissions)
                           }
-                          enabled={isExistingRule && (rule?.enabled ?? false)}
+                          enabled={isRuleSwitchActive}
                           startMlJobsIfNeeded={startMlJobsIfNeeded}
                           onChange={handleOnChangeEnabledRule}
+                          aria-label={
+                            rule?.name
+                              ? i18n.ruleSwitchAriaLabel(rule.name, isRuleSwitchActive)
+                              : undefined
+                          }
```

Comment:
Could we please avoid duplicating the `ruleSwitchAriaLabel` translation and move the `aria-label` calculation inside the `RuleSwitch` component?

In order to do that, `RuleSwitch` would need to accept a `ruleName` property. This property should be optional and the switch should work even if it's not provided. Maybe we could also generate some meaningful `aria-label` even if it's not provided.

Let's also move the translations added in this PR to a new `x-pack/plugins/security_solution/public/detections/components/rules/rule_switch/translations.ts` file.

Reply:
@banderror Honestly not sure which option I like more. In current versions, yes, we have some duplicating, but it's inline with how other  EUI components looks. In your version, we pass an extra `ruleName`, which, in fact, the switch component shouldn't know.

The decision will be on your side, but I vote for the current one.